File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> map1;

        for(int i = 0; i < nums.size(); i++) {
            map1[nums[i]] = i;
        }

        for(int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];
            if(map1.find(diff) != map1.end() && map1[diff] != i) {
                return vector<int>{i, map1[diff]};
            }
        }

        return vector<int>{-1, -1};
    }
};

File: Data Structures & Algorithms/first-missing-positive/submission-2.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int n = nums.length; 
        int[] a = new int[n + 1];
        int k = 0;

        for(int i = 0; i < n; i++) {
            int num = nums[i];
            if(num <= n && num >= 0) {
                a[num] = num;
            }
        }


        for(k = 1; k <= n; k++) {
            if(a[k] == 0) return k;
        }

        return k;
    }
}